Xamarin.Forms.Platform.Android/AppCompat/FormsAppCompatActivity.cs

```diff
@@ -180,19 +182,61 @@ protected override void OnCreate(Bundle savedInstanceState)
 			OnStateChanged();
 
 			AddStatusBarUnderlay();
+			Window.DecorView.ViewTreeObserver.GlobalLayout += ViewTreeObserverOnGlobalLayout;
 		}
 
 		protected override void OnDestroy()
 		{
 			// may never be called
 			base.OnDestroy();
 
+			Window.DecorView.ViewTreeObserver.GlobalLayout -= ViewTreeObserverOnGlobalLayout;
+
 			MessagingCenter.Unsubscribe<Page, AlertArguments>(this, Page.AlertSignalName);
 			MessagingCenter.Unsubscribe<Page, bool>(this, Page.BusySetSignalName);
 			MessagingCenter.Unsubscribe<Page, ActionSheetArguments>(this, Page.ActionSheetSignalName);
 
-			if (_platform != null)
-				_platform.Dispose();
+			_platform?.Dispose();
+		}
+
+		void ViewTreeObserverOnGlobalLayout(object sender, EventArgs eventArgs)
+		{
+			var rect = new Rect();
+			Window.DecorView.GetWindowVisibleDisplayFrame(rect);
+
+			int height = Window.DecorView.Context.Resources.DisplayMetrics.HeightPixels;
+			int diff = height - rect.Bottom;
+
+			if (diff != 0)
+			{
+				// set status bar color so status controls have a background
+				if (Forms.IsLollipopOrNewer)
+					Window.SetStatusBarColor(((ColorDrawable)_statusBarUnderlay.Background).Color);
+
+				if (Window.Attributes.SoftInputMode != SoftInput.AdjustResize)
+					return;
+
+				if (_layout.PaddingBottom != diff)
+				{
+					// keyboard has appeared in resize mode - set padding to show full height of the content
+					_layout.SetPadding(0, 0, 0, diff);
```

Comment:
Now `AdjustResize` has all the content we need even in fullscreen mode. If you had a `ScrollView` with 20 items and you tapped entry 18, you should be able to scroll to the top and bottom freely.